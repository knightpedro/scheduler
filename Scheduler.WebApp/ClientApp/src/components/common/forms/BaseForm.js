import React from 'react';
import { Formik, Form } from 'formik';
import Button from 'react-bootstrap/Button';

const BaseForm = ({ initialValues, validationSchema, onSubmit, onCancel, ...props }) => {
    return (
        <Formik 
            initialValues={initialValues}
            validationSchema={validationSchema}
            onSubmit={onSubmit}
            onCancel={onCancel}
        >
            {({ isSubmitting }) => (
                <Form>
                    {props.children}
                    <Button type="submit" className="mr-2" disabled={isSubmitting} variant="primary">Save</Button>
                    <Button variant="danger" onClick={onCancel}>Cancel</Button>
                </Form>
            )}
        </Formik>
    );
};

export default BaseForm