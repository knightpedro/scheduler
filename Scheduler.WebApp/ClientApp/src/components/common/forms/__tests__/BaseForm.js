import React from "react";
import { mount } from "enzyme";
import BaseForm from "../BaseForm";
import { Formik } from "formik";

describe("BaseForm", () => {
    it("renders correctly", () => {
        const wrapper = mount(<BaseForm />);
        expect(wrapper).toMatchSnapshot();
    });

    it("calls the submit handler", () => {
        const mockHandler = jest.fn();
        const wrapper = mount(<BaseForm onSubmit={mockHandler} />);
        wrapper
            .find(Formik)
            .props()
            .onSubmit();
        expect(mockHandler).toHaveBeenCalledTimes(1);
    });

    it("calls the cancel handler", () => {
        const mockHandler = jest.fn();
        const wrapper = mount(<BaseForm onCancel={mockHandler} />);
        wrapper
            .find("button")
            .last()
            .props()
            .onClick();
        expect(mockHandler).toHaveBeenCalledTimes(1);
    });
});
