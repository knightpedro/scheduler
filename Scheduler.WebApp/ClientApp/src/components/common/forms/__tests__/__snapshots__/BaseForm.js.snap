// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseForm renders correctly 1`] = `
<BaseForm>
  <Formik>
    <Form>
      <form
        action="#"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <Button
          active={false}
          className="mr-2"
          disabled={false}
          type="submit"
          variant="primary"
        >
          <button
            className="mr-2 btn btn-primary"
            disabled={false}
            type="submit"
          >
            Save
          </button>
        </Button>
        <Button
          active={false}
          disabled={false}
          type="button"
          variant="danger"
        >
          <button
            className="btn btn-danger"
            disabled={false}
            type="button"
          >
            Cancel
          </button>
        </Button>
      </form>
    </Form>
  </Formik>
</BaseForm>
`;
